controllers/library.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from models.library_items.items import LibraryItem
+from models.patrons.patron import Patron
+from pydantic import BaseModel
+from typing import Dict, List
+from dal.dal import insert_document_dal, delete_document_dal
+from utils.utils import patron_pydantic_to_mongoengine, item_pydantic_to_mongoengine
+from mongoengine import DoesNotExist
+from utils.custom_errors import *
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+"""
+Represents a library system with various attributes for managing its system.
+
+Attributes:
+    library_items (Dict[str, LibraryItem]): Dictionary of library items.
+    patrons (Dict[str, Patron]): Dictionary of library patrons.
+    bills (Dict[str, float]): Dictionary of patron bills.
+    name (str): The name of the library.
+"""
+
+
+class Library(BaseModel):
+    library_items: Dict[str, LibraryItem] = {}
+    patrons: Dict[str, Patron] = {}
+    bills: Dict[str, float] = {}  # Dictionary to keep track of all bills, keyed by patron ID
+    name: str
+
+    def add_new_library_items_to_the_library(self, new_library_items: List[LibraryItem]):
+        """
+        Add new library items to the library system.
+
+        Args:
+            new_library_items (List[LibraryItem]): List of library items to add.
+        """
+        for new_library_item in new_library_items:
+            if new_library_item.isbn in self.library_items.keys():
+                raise NotUniqueLibraryItemError(
```

Comment:
rename to AlreadyExists error and make generic. You can have a PatronAlreadyExists and a ItemAlreadyExists